### PR TITLE
Add volume mounts for test container

### LINKED ISSUE #35864: Property to mount a container volume during container (integration) testing
### Description

DISCLAIMER: I am not sure if this I technical possible with all the native stuff and so on. 

I have an application that access the file system to read/write files. Testing with `@QuarkusTest` is a piece of cake as all the data is on my local fs. When It comes to containerized native integration testing this gets a bit messy. The solution I currently use is an extra API which can create  or read testdata inside of the container. 
This means I have code for testing in `main` which already is definitly not good practice. I need to secure the API in production mode, so it is not callable anymore. This is error prone and poses a risk.

So it would be nice if I could write the files directly into the container. Or into a volume mount.

### Implementation ideas

In my imagination this would be configurable somehow like this:
`quarkus.test.container.volume-mounts."C:\dev\projects\myproject"=\home\myContainerFolder`
which would lead to the docker container under test is started with `-v C:\dev\projects\myproject:\home\myContainerFolder`
Now I still can create all my testdata on my local fs and everything works as expected without any extra messy hacking.

### PR DESCRIPTION
**Pull request by HerrDerb**
closes https://github.com/quarkusio/quarkus/issues/35864

### CODE REVIEW

**Comment by geoand** [comment]
Very nice!

One note - it should be `volume` everywhere instead of `volumn`.

As for [your comment](https://github.com/quarkusio/quarkus/issues/35864#issuecomment-1716589007)  about warning, the problem is that you are setting the wrong property :).
If you use something like:

```
mvn test -Dquarkus.test.container.volume-mounts."/test"=/home/test
```

everything works as expected

**Comment by HerrDerb** [comment]
Thanks for having a look👍Oh I see, I would find it more natural to set the property the same way as you define a volumne mount in docker 🤔
But I guess you already did it that way with the port bindings.

**Comment by geoand** [comment]
Yeah, it's more natural for Quarkus as that it the way named things are generally configured

File: core/deployment/src/main/java/io/quarkus/deployment/dev/testing/TestConfig.java [hunk 1]
```diff
@@ -299,6 +299,12 @@ public static class Container {
          */
         @ConfigItem
         Map<String, String> labels;
+
+        /**
+         * A set of volume mounts to add to the launched container (localDirectory=containerDirectory)
```
Comment (geoand) [inline core/deployment/src/main/java/io/quarkus/deployment/dev/testing/TestConfig.java:304]:
```suggestion
         * A set of volume mounts to add to the launched container
```

**Comment by HerrDerb** [comment]
Finally I could produce a positiv run too :) Thanks for you help!

**Comment by geoand** [comment]
🎉 

When you done, please squash the commits, then I think we're good to go

**Comment by HerrDerb** [comment]
Done 👍

**Review by geoand — APPROVED** [review]

**Pull request merged**